core/src/main/kotlin/in/specmatic/conversions/OpenApiSpecification.kt

```diff
@@ -627,6 +627,7 @@ class OpenApiSpecification(private val openApiFile: String, val openApi: OpenAPI
                     throw UnsupportedOperationException("Unsupported composed schema: $schema")
                 }
             }
+            is ByteArraySchema -> BinaryPattern()
```

Comment:
We'd need to implement a new Pattern class for ByteArraySchema here. See my comment at https://github.com/znsio/specmatic/pull/716#issuecomment-1722482596.

Reply:
Yes, i came to the same conclusion while working with this quick fix: it needs a new Pattern with it's own matching logic & generation.

It would be cool to be able to use externalData for stubbing with this Pattern Class (taking the file bytes and encoding them to base64)
Currently if i understand correctly: 

When dealing with externalData, the parser adds _filename to the key that corresponds to the file content, then somewhere in the response generation, the file content is added to the response without looking at the pattern type.

Maybe we'll need to adapt this behaviour to support different externalData treatment based on the PatternType?

Reply:
After looking further, i realized that externalValue is not supported for the responses yet. i did not find anything that loads file content in the response. This could be useful if we want to send a string of bytes or just return an application octet stream, i'm gonna create an issue for this